paasta_tools/contrib/paasta_update_soa_memcpu.py

```diff
@@ -75,17 +59,51 @@ def in_tempdir():
             yield
 
 
+def get_perf_data(creds, filename):
+    url = 'https://splunk-api.yelpcorp.com/servicesNS/nobody/yelp_performance/search/jobs/export'
+    search = (
+        '| inputlookup {} |'
+        ' eval _time = search_time | where _time > relative_time(now(),\"-7d\")'
+    ).format(filename)
+    data = {
+        'output_mode': 'json',
+        'search': search,
+    }
+    creds = creds.split(':')
+    resp = requests.post(url, data=data, auth=(creds[0], creds[1]))
+    resp_text = resp.text.split('\n')
+    resp_text = [x for x in resp_text if x]
+    resp_text = [json.loads(x) for x in resp_text]
+
+    services_to_update = []
+    for d in resp_text:
+        criteria = d['result']['criteria'].split()
+        serv = {}
+        serv['service'] = criteria[0]
+        serv['cluster'] = criteria[1]
+        serv['instance'] = criteria[2]
+        serv['cpus'] = d['result']['suggested_cpus']
+        serv['owner'] = d['result']['service_owner']
+        serv['money'] = d['result']['estimated_monthly_savings']
+        serv['date'] = d['result']['_time'].split(" ")[0]
+        serv['old_cpus'] = d['result']['current_cpus']
+        serv['project'] = d['result']['project']
+        services_to_update.append(serv)
+
+    return services_to_update
+
+
 def clone(branch_name):
     print('cloning')
     remote = 'git@sysgit.yelpcorp.com:yelpsoa-configs'
     subprocess.check_call(('git', 'clone', remote, '.'))
-    subprocess.check_call(('git', 'checkout', 'origin/master', '-b', branch_name))
+    subprocess.check_call(('git', 'checkout', '-b', branch_name))
 
 
-def commit(filename, memcpu):
-    message = 'Updating {} for under/overprovisioned {}'.format(filename, memcpu)
+def commit(filename, over):
+    message = 'Updating {} for {}provisioned cpu'.format(filename, over)
```

Comment:
Can you make this more descriptive? Heck have the whole message that was put into the reviewboard description and stick it in here! you can never not have enough context.

Reply:
The entire reviewboard description would be many lines for a commit message. I don't want to make the git history too hard to read. I'll add a little more detail though.